src/main/java/com/github/manolo8/darkbot/gui/players/PlayerEditor.java

```diff
@@ -64,6 +91,59 @@ public void refreshList(String filter) {
                 .filter(pi -> pi.filter(query))
                 .sorted(Comparator.comparing(pi -> pi.username))
                 .forEach(playersModel::addElement);
+        renderer.setAddedPlayerCount(playersModel.getSize());
+        for (PlayerInfo p : nearbyPlayers) {
+            playersModel.addElement(p);
+        }
+    }
+
+    @EventHandler
+    public void onEntityPlayerAdd(EntitiesAPI.EntityCreateEvent event) {
+        if (event.getEntity() instanceof Player
+                && playersModel != null
+                && main.config.PLAYER_INFOS != null
+                && !main.config.PLAYER_INFOS.containsKey(event.getEntity().getId())) {
+
+            SwingUtilities.invokeLater(() -> {
+                int idx = indexOfPlayersModel(event.getEntity().getId());
+                if (idx < 0) {
+                    PlayerInfo playerInfo = new PlayerInfo((Player) event.getEntity());
+                    nearbyPlayers.add(playerInfo);
+                    playersModel.addElement(playerInfo);
+                } else {
+                    PlayerInfo info = playersModel.get(idx);
+                    nearbyPlayers.add(info);
+                }
+            });
+        }
+    }
+
+    @EventHandler
+    public void onEntityPlayerRemove(EntitiesAPI.EntityRemoveEvent event) {
+        if (event.getEntity() instanceof Player
+                && playersModel != null
+                && main.config.PLAYER_INFOS != null
+                && !main.config.PLAYER_INFOS.containsKey(event.getEntity().getId())) {
+
+            SwingUtilities.invokeLater(() -> {
+                int idx = indexOfPlayersModel(event.getEntity().getId());
+                if (idx != -1) {
+                    PlayerInfo info = playersModel.getElementAt(idx);
+                    if (!playerInfoList.isSelectedIndex(idx)) {
+                        nearbyPlayers.remove(info);
+                        playersModel.removeElement(info);
+                    } else {
+                        nearbyPlayers.remove(info);
+                    }
+                }
```

Comment:
```suggestion
                if (idx == -1) return;
                PlayerInfo info = playersModel.getElementAt(idx);
                if (!playerInfoList.isSelectedIndex(idx)) {
                    nearbyPlayers.remove(info);
                    playersModel.removeElement(info);
                } else {
                    nearbyPlayers.remove(info);
                }
```
Less indenting makes the code easier to follow